### PR TITLE
test: Add .NET contract tests

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## References

* #1023

### CODE REVIEW

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/Mappers/SettlementMethodMapper.cs [hunk 1]
```diff
@@ -25,6 +25,10 @@ public static class SettlementMethodMapper
             TimeSeriesType.FlexConsumption => SettlementMethod.Flex,
             TimeSeriesType.NonProfiledConsumption => SettlementMethod.NonProfiled,
             TimeSeriesType.NetExchangePerGa => null,
+            TimeSeriesType.NetExchangePerNeighboringGa => null,
+            TimeSeriesType.GridLoss => null,
+            TimeSeriesType.NegativeGridLoss => null,
+            TimeSeriesType.PositiveGridLoss => SettlementMethod.Flex,
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/Mappers/SettlementMethodMapper.cs:31]: why does `PositiveGridLoss` equal Flex?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/DeltaTableContracts/QuantityQualityMapperTests.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            // Act
+            var actual = QuantityQualityMapper.FromDeltaTableValue(validDeltaValue);
+
+            // Assert it's a defined enum value (and not null)
+            actual.Should().BeDefined();
+        }
+    }
+
+    [Theory]
+    [InlineAutoMoqData("calculated", QuantityQuality.Calculated)]
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/DeltaTableContracts/QuantityQualityMapperTests.cs:51]: Maybe a comment on where these "magic" strings are coming from, would help the non-python devs :) (Or maybe I'm missing something completely)

**Review by lasrinnil — APPROVED** [review]

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/Mappers/SettlementMethodMapper.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/Mappers/SettlementMethodMapper.cs:31]: Positive grid loss is consumption and according to SME it's flex consumption.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/DeltaTableContracts/QuantityQualityMapperTests.cs (hunk 2 above)
Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/DeltaTableContracts/QuantityQualityMapperTests.cs:51]: I'm open for suggestions 😄 
The tests are in the "DeltaTableContracts" namespace/folder and the magic strings are named "deltaValue" in the test method parameters. That was my attempt of telling that these are the values you should expect when reading from the delta table.

**Pull request merged**